frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -760,19 +772,23 @@ QMenu *OBSBasic::CreatePerSceneTransitionMenu()
 	duration->setValue(curDuration);
 
 	auto setTransition = [this](QAction *action) {
-		int idx = action->property("transition_index").toInt();
+		std::string uuid = action->property("transition_uuid").toString().toStdString();
 		OBSSource scene = GetCurrentSceneSource();
 		OBSDataAutoRelease data = obs_source_get_private_settings(scene);
+		OBSSource transition;
 
-		if (idx == -1) {
+		if (uuid.empty() || transitions.find(uuid) == transitions.end()) {
 			obs_data_set_string(data, "transition", "");
 			return;
 		}
 
-		OBSSource tr = GetTransitionComboItem(ui->transitions, idx);
+		if (transitions.find(uuid) == transitions.end())
+			return;
 
-		if (tr) {
-			const char *name = obs_source_get_name(tr);
+		transition = transitions[uuid];
```

Comment:
```suggestion
auto foundTransition = transitions.find(uuid);

if (foundTransition == transitions.end()) {
    return;
}

transition = foundTransition->second;
```
See above - use iterator to access found element instead of using mutating subscript operator.

Reply:
Mutators removed